feat(env-options): env-options conveniences for common cases (#1710)
endojs/endo: packages/env-options/README.md

```js
import { getEnvironmentOption } from '@endo/env-options';
const FooBarOption = getEnvironmentOption('FOO_BAR', 'absent');
```

The first argument to `getEnvironmentOption` is the name of the option.
The value of `FooBarOption` would then be the value of
`globalThis.process.env.FOO_BAR`, if present.
If value is either absent or `undefined`, the second argument,
such as `'absent'`, would be used instead.

In either case, reflecting Unix environment variable expectations,
the resulting setting must be a string.
This restriction also helps ensure that this channel is used only to pass data,
not authority beyond the ability to read this global state.

```js
const ENABLED =
  getEnvironmentOption('TRACK_TURNS', 'disabled', ['enabled']) === 'enabled';
```

`getEnvironmentOption` also takes an optional third argument, which if present
is an exhaustive list of allowed strings other than the default. If present
and the actual environment option is neither the default nor one of these
allowed strings, then an error is thrown explaining the problem.

```js
const DEBUG_VALUES = getEnvironmentOptionsList('DEBUG');
const DEBUG_AGORIC = environmentOptionsListHas('DEBUG', 'agoric');
```

> Another common Unix convention is for the value of an option to be a

Reviewer: ```suggestion
Another common convention is for the value of an option to be a
```

Author: Done

comma (`','`) separated list of strings. `getEnvironmentOptionsList` will

> return this list, or an empty list of the option is absent.

Reviewer: ```suggestion
return this list, or an empty list if the option is absent.
```

Author: Done

`environmentOptionsListHas` will test if this list contains a specific
value, or return false if the option is absent.

(Compat note: https://github.com/Agoric/agoric-sdk/issues/8096 explains that
for `DEBUG` specifically, some existing uses split on colon (`':'`) rather
than comma. Once these are fixed, then these uses can be switched to use
`getEnvironmentOptionsList` or `environmentOptionsListHas`.)

## Tracking used option names

The `'@endo/env-options'` module also exports a lower-level
`makeEnvironmentCaptor` that you can apply to whatever object you wish to treat

> as a global, such as the global of another compartment. It returns an entagled

Reviewer: ```suggestion
as a global (having a "process" property with its own "env" record),
such as the global of another compartment. It returns an entangled
```

Author: Done

pair of a `getEnvironmentOption` function as above, and a

> `getCapturedEnvironmentOptionNames` function for that returns an array of

Reviewer: ```suggestion
`getCapturedEnvironmentOptionNames` function that returns an array of
```

Author: Done
